pipelines/update_raster_stats.py

```diff
@@ -0,0 +1,148 @@
+from datetime import datetime
+
+import pandas as pd
+import xarray as xr
+from tqdm.auto import tqdm
+
+from src.constants import ISO3S
+from src.datasources import codab
+from src.utils import blob, database
+
+
+def get_existing_stats_dates(iso3: str, engine) -> list:
+    query = f"""
+    SELECT DISTINCT valid_date
+    FROM app.floodscan_exposure
+    WHERE iso3 = '{iso3.upper()}'
+    ORDER BY valid_date
+    """
+    df_unique_dates = pd.read_sql(query, con=engine)
+    df_unique_dates["valid_date"] = pd.to_datetime(
+        df_unique_dates["valid_date"]
+    )
+    return df_unique_dates["valid_date"].to_list()
+
+
+if __name__ == "__main__":
+    engine = database.get_engine(stage="dev")
+    clobber = False
+    verbose = False
+
+    for iso3 in ISO3S:
+        print(f"Processing {iso3}")
+        adm = codab.load_codab_from_blob(iso3, admin_level=2)
+        existing_exposure_rasters = [
+            x
+            for x in blob.list_container_blobs(
+                name_starts_with=f"{blob.PROJECT_PREFIX}/processed/"
+                f"flood_exposure/{iso3}/"
+            )
+            if x.endswith(".tif")
+        ]
+        existing_dates = get_existing_stats_dates(iso3, engine)
+        unprocessed_exposure_rasters = [
```

Comment:
It would be useful to have some logging here to give a sense of how many unprocessed rasters need to be processed.

Reply:
Yup totally #11